homeassistant/components/hvac/zwave.py

```diff
@@ -0,0 +1,231 @@
+                    _LOGGER.debug('Found Remotec ZXT-120 Zwave Thermostat')
+                    self._zxt_120 = 1
+
+    def value_changed(self, value):
+        """Called when a value has changed on the network."""
+        if self._value.node == value.node:
+            self.update_properties()
+            self.update_ha_state(True)
+            _LOGGER.debug('Value changed on network %s)', value)
+
+    def update_properties(self):
+        """Callback on data change for the registered node/value pair."""
+        # Set point
+        for _, value in self._node.get_values(class_id=0x43).items():
+            if int(value.data) != 0:
+                self._target_temperature = int(value.data)
+        # Operation Mode
+        for _, value in self._node.get_values(class_id=0x40).items():
+            self._current_operation = value.data
+            self._operation_list = list(value.data_items)
+            _LOGGER.debug('self._operation_list=%s)', self._operation_list)
+        # Current Temp
+        for _, value in self._node.get_values(class_id=0x31).items():
+            self._current_temperature = int(value.data)
+            self._unit = value.units
+        # Fan Mode
+        if self._zxt_120 == 1:
+            for _, value in self._node.get_values(class_id=0x44).items():
+                self._current_operation_state = value.data
+                self._fan_list = list(value.data_items)
+                _LOGGER.debug('self._fan_list=%s)', self._fan_list)
+            for value in self._node.get_values(class_id=0x70).values():
+                if value.command_class == 112 and value.index == 33:
+                    self._current_swing_mode = value.data
+                    self._swing_list = [0, 1]
+                    _LOGGER.debug('self._swing_list=%s)', self._swing_list)
+
+        else:
+            for _, value in self._node.get_values(class_id=0x42).items():
+                self._current_operation_state = value.data
+                self._fan_list = list(value.data_items)
+                _LOGGER.debug('self._fan_list=%s)', self._fan_list)
+
+    @property
+    def should_poll(self):
+        """No polling on ZWave."""
+        return False
+
+    @property
+    def current_fan_mode(self):
+        """Return the fan speed set."""
+        return self._current_operation_state
+
+    @property
+    def fan_list(self):
+        """List of available fan modes."""
+        return self._fan_list
+
+    @property
+    def current_swing_mode(self):
+        """Return the swing mode set."""
+        return self._current_swing_mode
+
+    @property
+    def swing_list(self):
+        """List of available swing modes."""
+        return self._swing_list
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement."""
+        unit = self._unit
+        if unit == 'C':
+            return TEMP_CELSIUS
+        elif unit == 'F':
+            return TEMP_FAHRENHEIT
+        else:
+            return unit
```

Comment:
Instead of just returning the unit, better to raise an exception with unexpected unit.